show/system_health.py

```diff
@@ -159,3 +166,76 @@ def sysready_status_detail():
         clicommon.run_command(cmd, display_cmd=False)
     except Exception as e:
         click.echo("Exception: {}".format(str(e)))
+
+
+def show_module_state(module_name):
+    chassis_state_db = SonicV2Connector(host=CHASSIS_SERVER, port=CHASSIS_SERVER_PORT)
+    chassis_state_db.connect(chassis_state_db.CHASSIS_STATE_DB)
+    key = 'DPU_STATE|'
+    suffix = '*' if not module_name or not module_name.startswith("DPU") else module_name
+    key = key + suffix
+    keys = chassis_state_db.keys(chassis_state_db.CHASSIS_STATE_DB, key)
+    if not keys:
+        return
+
+    table = []
+    for dbkey in natsorted(keys):
+        key_list = dbkey.split('|')
+        if len(key_list) != 2:  # error data in DB, log it and ignore
+            continue
+        state_info = chassis_state_db.get_all(chassis_state_db.CHASSIS_STATE_DB, dbkey)
+        # Determine operational status
+        # dpu_states = [value for key, value in state_info.items() if key.endswith('_state')]
+
+        midplanedown = False
+        up_cnt = 0
+        for key, value in state_info.items():
+            if key.endswith('_state'):
+                if value.lower() == 'up':
+                    up_cnt = up_cnt + 1
+                if 'midplane' in key and value.lower() == 'down':
+                    midplanedown = True
```

Comment:
@rameshraghupathy can  you explain whats going on here? Either the midplane is down or up when the user executed the CLI. ?

Reply:
Marking the exact granular state of the DPU.
Oper-Status definition: 
Online : All states are up
Offline: dpu_midplane_link_state is down
Partial Online: dpu_midplane_link_state is up and dpu_control_plane_state or dpu_data_plane_state is down